code/modules/mob/living/carbon/xenomorph/XenoProcs.dm

```diff
@@ -747,3 +747,16 @@
 /// Handler callback to reset immobilization status after a successful [/mob/living/carbon/xenomorph/proc/throw_carbon]
 /mob/living/carbon/xenomorph/proc/throw_carbon_end(mob/living/carbon/target)
 	REMOVE_TRAIT(target, TRAIT_IMMOBILIZED, XENO_THROW_TRAIT)
+
+/// snowflake proc to clear effects from research warcrimes
+/mob/living/carbon/xenomorph/proc/clear_debuffs()
+	var/debuffs = list(/datum/component/status_effect/healing_reduction, /datum/component/status_effect/toxic_buildup, /datum/component/status_effect/speed_modifier)
+	for(var/datum/component/status_effect/debuff as anything in debuffs)
+		var/datum/component/status_effect/instance = GetComponent(debuff)
+		if(instance)
+			instance.cleanse()
```

Comment:
Instead of this, instead send a signal that all `/datum/component/status_effect` register to so then cleanse() is the SIGNAL_HANDLER for.

Reply:
To clarify, this would be a new signal - not any existing signal.